impl/src/main/java/org/ehcache/config/builders/ConfigurationBuilder.java

```diff
@@ -73,7 +75,8 @@ private ConfigurationBuilder(ConfigurationBuilder builder, ClassLoader classLoad
 
   @Override
   public Configuration build() {
-    return new DefaultConfiguration(caches, classLoader, serviceConfigurations.toArray(new ServiceCreationConfiguration<?>[serviceConfigurations.size()]));
+    List<ServiceCreationConfiguration<?>> serviceCreationConfigurations = addDiscoveredServiceConfiguration();
```

Comment:
Weird to return a list just to go to an array right after

Reply:
How you would do it more elegantly ? The list not being modifiable, we have to recreate it :-(

Reply:
@SuppressWarnings("unchecked")
  public ServiceCreationConfiguration<?>[] addDiscoveredServiceConfiguration() {
    return Stream.concat(serviceConfigurations.stream(), 
      StreamSupport.stream(ServiceLoader.load(ServiceCreationConfigurationProvider.class, classLoader).spliterator(), false)
      .filter(provider -> findServiceByClass(provider.getServiceType()) == null)
      .map((Function<ServiceCreationConfigurationProvider, Object>) Supplier::get))
      .toArray(ServiceCreationConfiguration[]::new);
  }